frontend/src/employee-mobile-frontend/child-attendance/actions/MarkReservations.tsx

```diff
@@ -0,0 +1,540 @@
+    (maxDate, reservation) =>
+      maxDate === null || reservation.date.isAfter(maxDate)
+        ? reservation.date
+        : maxDate,
+    null
+  )
+  const laterAbsences =
+    maxDate !== null
+      ? absences.filter((absence) => absence.date.isAfter(maxDate))
+      : absences
+
+  return (
+    <>
+      {reservations.map((reservation) => (
+        <React.Fragment key={reservation.date.format()}>
+          {minDatesByWeek[reservation.date.getIsoWeek()] ===
+            reservation.date && (
+            <Label primary>
+              {i18n.common.week} {reservation.date.getIsoWeek()}
+            </Label>
+          )}
+          <FixedSpaceRow
+            alignItems="baseline"
+            data-qa={`reservation-date-${reservation.date.formatIso()}`}
+          >
+            <Label>{reservation.date.format('EEEEEE d.M.', lang)}</Label>
+            <ReservationView reservation={reservation} />
+          </FixedSpaceRow>
+          <Gap size="s" />
+        </React.Fragment>
+      ))}
+      {laterAbsences.length > 0 ? (
+        <>
+          <HorizontalLine slim dashed />
+          <Label primary>{i18n.absences.laterAbsence}</Label>
+          {groupAbsencesByDateRange(laterAbsences).map((absenceRange) => {
+            if (absenceRange.durationInDays() > 1) {
+              return (
+                <div key={absenceRange.start.format()} data-qa="absence-row">
+                  {`${absenceRange.start.format()} - ${absenceRange.end.format()}`}
+                </div>
+              )
+            } else {
+              return (
+                <div key={absenceRange.start.format()} data-qa="absence-row">
+                  {absenceRange.start.format()}
+                </div>
+              )
+            }
+          })}
+        </>
+      ) : null}
+      <HorizontalLine slim />
+      <Actions>
+        <FixedSpaceRow fullWidth>
+          <Button
+            text={
+              i18n.attendances.actions.nonReservableReservations
+                .markReservations
+            }
+            onClick={onEditReservations}
+            data-qa="edit"
+          />
+          <Button
+            primary
+            text={
+              i18n.attendances.actions.nonReservableReservations
+                .markAbsentBeforehand
+            }
+            onClick={onMarkAbsence}
+            data-qa="mark-absent-beforehand"
+          />
+        </FixedSpaceRow>
+      </Actions>
+    </>
+  )
+}
+
+const ReservationView = ({
+  reservation
+}: {
+  reservation: NonReservableReservation
+}) => {
+  const { i18n } = useTranslation()
+
+  return (
+    <div data-qa="reservation-text">
+      {reservation.absenceType !== null ? (
+        <ServiceTime>{i18n.attendances.types.ABSENT}</ServiceTime>
+      ) : (
+        <AttendanceDailyServiceTimes
+          hideLabel={true}
+          times={reservation.dailyServiceTimes}
+          reservations={reservation.reservations}
```

Comment:
Reservations are shown in the wrong order now if the user enters them in the wrong order.

Reply:
Fixed